multinode_integration_tests/src/mock_blockchain_client_server.rs

```diff
@@ -1,397 +1,22 @@
-                match HTTP_VERSION_DETECTOR.captures(&conn_state.request_accumulator) {
-                    Some(captures) => {
-                        let http_version = captures.get(1).unwrap().as_str();
-                        if http_version != "1.1" {
-                            panic!("MBCS handles only HTTP version 1.1, not {}", http_version)
-                        }
-                    }
-                    None => panic!("Request has no HTTP version"),
-                }
-                match CONTENT_LENGTH_DETECTOR.captures(&conn_state.request_accumulator) {
-                    Some(captures) => {
-                        let content_length =
-                            captures.get(1).unwrap().as_str().parse::<usize>().unwrap();
-                        conn_state.request_stage = RequestStage::Parsed {
-                            content_offset,
-                            content_length,
-                        };
-                        None
-                    }
-                    None => panic!("Request has no Content-Length header"),
-                }
-            }
-        }
-    }
-
-    fn handle_parsed(
-        conn_state: &mut ConnectionState,
-        content_offset: usize,
-        content_length: usize,
-    ) -> Option<String> {
-        let request_length = content_offset + content_length;
-        if conn_state.request_accumulator.len() >= request_length {
-            let request = conn_state.request_accumulator[0..request_length].to_string();
-            // TODO: What happens if we have multibyte Unicode characters here?
-            let delete_count = request.len();
-            let remaining_len = conn_state.receive_buffer_occupied - delete_count;
-            for i in 0..remaining_len {
-                conn_state.receive_buffer[i] = conn_state.receive_buffer[i + delete_count];
-            }
-            conn_state.receive_buffer_occupied -= delete_count;
-            conn_state.request_accumulator =
-                conn_state.request_accumulator[delete_count..].to_string();
-            conn_state.request_stage = RequestStage::Unparsed;
-            Some(request)
-        } else {
-            None
-        }
-    }
-
-    fn send_body(conn_state: &mut ConnectionState, response: String) {
-        let http = format!(
-            "HTTP/1.1 200 OK\r\nContent-Type: application/json\r\nContent-Length: {}\r\n\r\n{}",
-            response.len(),
-            response
-        );
-        conn_state.conn.write_all(http.as_bytes()).unwrap();
-    }
-}
-
-#[derive(PartialEq, Eq, Clone, Debug)]
-enum RequestStage {
-    Unparsed,
-    Parsed {
-        content_offset: usize,
-        content_length: usize,
-    },
-}
-
-struct ConnectionState {
-    conn: TcpStream,
-    receive_buffer: [u8; 16384],
-    receive_buffer_occupied: usize,
-    request_stage: RequestStage,
-    request_accumulator: String,
-}
+// Code has been migrated to masq_lib/src/test_utils/mock_blockchain_client_server.rs
```

Comment:
Maybe saying: `// The actual mock server has been migrated to ...` and also please find the number of the card which is planned to take care of moving these tests down here also towards the mock server and mention it here in a TODO.